terminus-ui/cohort-date-range/src/cohort-date-range.component.md

```diff
@@ -0,0 +1,95 @@
+<h1>Cohort Date Range</h1>
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**
+
+- [Basic usage](#basic-usage)
+- [Event driven](#event-driven)
+- [Inputs](#inputs)
+  - [allowCustomDates](#allowCustomDates)
+  - [Disable the component](#disable-the-component)
+
+## Basic usage
+
+Pass in the cohorts:
+
+```html
+<ts-cohort-date-range
+  [cohorts]="cohorts"
+></ts-cohort-date-range>
+```
+
+```typescript
+  public cohorts = [{
+    display: 'Last 90 days',
+    range: {
+      start: startOfDay(subDays(new Date(), 90)),
+      end: currentDate,
+    },
+  }, {
+    display: 'Last full month',
+    range: {
+      start: startOfDay(subMonths(startOfMonth(currentDate), 1)),
+      end: endOfDay(subDays(startOfMonth(currentDate), 1)),
+    },
+  }, {
+    display: 'Custom dates',
+    range: {
+      start: '',
+      end: '',
+    },
+  }];
+```
+
+NOTE: The keys inside the passed in `cohorts` object has to be in the form of
+```typescript
+{ 
+  display: string,
+  range: {
+   start: Date | string,
+   end: Date | string,
+   }
+}
+```
+
+NOTE: For any custom dates which allows user to select a date from date 
+range selection would need to pass in start and end date with empty string:
+```typescript
+{
+  display: 'Custom Dates',
+  range: {
+    start: '',
+    end: '',
+  }
+}
+```
+
+## Event driven
+
+The event that's tied into this component is `cohortDateRangeChanged`.
+
+```html
+<ts-cohort-date-range
+  [cohorts]="cohorts"
+  (cohortDateRangeChanged)="printRange($event)"
+></ts-cohort-date-range>
+```
+
+`cohortDateRangeChanged` is fired off when the date range changes.
+
+## Inputs to the component
+
+### allowCustomDates
+
+`allowCustomDates` defaults to `false`. When set to `false`, date range is readonly.
```

Comment:
We should add code snippet to anything we think is important enough to call out in usage docs.